src/components/ValueSet/ValueSetForm.tsx

```diff
@@ -319,14 +337,29 @@ export function ValueSetForm({
         z.object({
           system: z.string(),
           concept: z
-            .array(z.object({ code: z.string(), display: z.string() }))
+            .array(
+              z.object({
+                code: z.string().refine((val) => val !== "", {
+                  message: t("code_required"),
+                }),
+                display: z.string().refine((val) => val !== "", {
+                  message: t("display_required"),
+                }),
+              }),
+            )
             .optional(),
           filter: z
             .array(
               z.object({
-                property: z.string(),
-                op: z.string(),
-                value: z.string(),
+                property: z.string().refine((val) => val !== "", {
+                  message: t("property_required"),
+                }),
```

Comment:
see how required fields are done in other places in care...

<img width="784" alt="image" src="https://github.com/user-attachments/assets/7272c89d-754b-4435-823c-50434ef9758e" />